### PR TITLE
Refactor example to modules

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Also simplify `connect_activate` calls

### CODE REVIEW

File: examples/cairo_threads/main.rs [hunk 1]
```diff
@@ -1,127 +1,41 @@
...
-//
-// The two images per thread are allocated and initialized once and sent back and forth repeatedly.
-
 fn build_ui(application: &gtk::Application) {
     let window = ApplicationWindow::new(application);
     let area = DrawingArea::new();
     window.add(&area);
 
-    area.set_size_request(WIDTH * 2, HEIGHT * 2);
+    let (width, height) = (200, 200);
```
Comment (sdroege) [inline examples/cairo_threads/main.rs:34]: Why replace these constants with hardcoded values here? Having them being passed to the functions as arguments makes sense, but in this function at least making direct use of constants seems cleaner.

File: examples/basics/main.rs [hunk 2]
```diff
@@ -22,9 +22,7 @@ fn main() {
         gtk::Application::new(Some("com.github.gtk-rs.examples.basic"), Default::default())
             .expect("Initialization failed...");
 
-    application.connect_activate(|app| {
-        build_ui(app);
-    });
+    application.connect_activate(build_ui);
```
Comment (sdroege) [inline examples/basics/main.rs:25]: Interesting, why didn't clippy warning about that?

File: examples/cairo_threads/main.rs (hunk 1 above)
Comment (Hofer-Julian) [reply examples/cairo_threads/main.rs:34]: Sure, I can readd the constant

File: examples/composite_template/example_application_window/mod.rs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        @extends gtk::Widget, gtk::Window, gtk::ApplicationWindow, @implements gio::ActionMap, gio::ActionGroup;
+}
+
+impl ExampleApplicationWindow {
+    pub fn new<P: glib::IsA<gtk::Application>>(app: &P) -> Self {
+        glib::Object::new(&[("application", app)])
+            .expect("Failed to create ExampleApplicationWindow")
+    }
+
+    pub fn init_label(&self) {
```
Comment (sdroege) [inline examples/composite_template/example_application_window/mod.rs:18]: This doesn't have to be public as it's only used by the `imp` submodule. Should never really be called from the outside

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise :)

File: examples/basics/main.rs (hunk 2 above)
Comment (Hofer-Julian) [reply examples/basics/main.rs:25]: Good question

**Comment by Hofer-Julian** [comment]
I think the PR can be merged now.

Since reviews are so fast, I will continue with the smaller PRs.

**Review by sdroege — APPROVED** [review]

**Comment by sdroege** [comment]
> I think the PR can be merged now.

If you want to make me more happy, squash the two fixup commits into the corresponding commits :) Otherwise all good from my side.

**Comment by Hofer-Julian** [comment]
> > I think the PR can be merged now.
> 
> If you want to make me more happy, squash the two fixup commits into the corresponding commits :) Otherwise all good from my side.

Done

**Review by sdroege — APPROVED** [review]

File: examples/cairo_threads/image/image_holder.rs [hunk 4]
```diff
@@ -0,0 +1,42 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+/// Helper struct that allows passing the pixels to the Cairo image surface and once the
```
Comment (GuillaumeGomez) [inline examples/cairo_threads/image/image_holder.rs:3]: Please add an empty line before the doc comment.

Comment (Hofer-Julian) [reply examples/cairo_threads/image/image_holder.rs:3]: Mmmh, too bad that rustfmt does not do that for me

Comment (GuillaumeGomez) [reply examples/cairo_threads/image/image_holder.rs:4]: Open an issue there? :)

**Review by GuillaumeGomez — APPROVED** [review]

File: examples/cairo_threads/image/image_holder.rs (hunk 4 above)
Comment (Hofer-Julian) [reply examples/cairo_threads/image/image_holder.rs:4]: Done (the line change not the issue)

Comment (GuillaumeGomez) [reply examples/cairo_threads/image/image_holder.rs:4]: Please open the issue. Everyone will benefit from it.

Comment (Hofer-Julian) [reply examples/cairo_threads/image/image_holder.rs:4]: Sure, will see first if it exists already

Comment (GuillaumeGomez) [reply examples/cairo_threads/image/image_holder.rs:4]: Thanks!

**Review by bilelmoussaoui — APPROVED** [review]

**Pull request merged**